src/Components/Entry/UploadFileForm.test.tsx

```diff
@@ -0,0 +1,327 @@
+import { mount, ReactWrapper } from "enzyme";
+import { mockJSDOM } from "../../Helpers/TestHelpers";
+import { Client } from "reduct-js";
+import EntryDetail from "../../Views/BucketPanel/EntryDetail";
+import { MemoryRouter } from "react-router-dom";
+import { act } from "react-dom/test-utils";
+import "codemirror/lib/codemirror.css";
+import "codemirror/mode/javascript/javascript";
+import { Modal, Button } from "antd";
+import fs from "fs";
+import path from "path";
+import { waitFor } from "@testing-library/react";
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useParams: () => ({
+    bucketName: "testBucket",
+    entryName: "testEntry",
+  }),
+}));
+
+describe("UploadFileForm", () => {
+  let wrapper: ReactWrapper;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.resetModules();
+
+    mockJSDOM();
+
+    wrapper = mount(
+      <MemoryRouter>
+        <EntryDetail client={new Client("")} />
+      </MemoryRouter>,
+    );
+  });
+
+  afterEach(() => {
+    if (wrapper) {
+      wrapper.unmount();
+    }
+  });
+
+  describe("UI Elements", () => {
+    beforeEach(async () => {});
+
+    it("should successfully upload with default values", async () => {
+      const onUploadSuccess = jest.fn();
+
+      const wrapper = mount(
+        <MemoryRouter>
+          <EntryDetail
+            client={new Client("")}
+            permissions={{ fullAccess: true }}
+          />
+        </MemoryRouter>,
+      );
+
+      const uploadButton = wrapper.find('button[title="Upload File"]');
+      expect(uploadButton.exists()).toBe(true);
+
+      await act(async () => {
+        uploadButton.simulate("click");
+      });
+      wrapper.update();
+
+      const uploadModal = wrapper
+        .find(Modal)
+        .filterWhere((modal) => modal.prop("title") === "Upload File");
+      expect(uploadModal.props().open).toBe(true);
+
+      const filePath = path.resolve(__dirname, "test_files", "test.txt");
+      const fileContent = fs.readFileSync(filePath, "utf-8");
+
+      const file = new File([fileContent], "test.txt", {
+        type: "text/plain",
+        lastModified: new Date().getTime(),
+      });
+
+      const uploadForm = uploadModal.find("UploadFileForm");
+      expect(uploadForm.exists()).toBe(true);
+
+      const fileInput = uploadForm.find('input[type="file"]');
+      expect(fileInput.exists()).toBe(true);
+
+      const reader = new FileReader();
+      reader.onload = async () => {
+        const dataTransfer = {
```

Comment:
Let's work on this test and then check others. 

This test doesn't test anything important because you don't wait for the `onload` event at the end of your test. 

It's a bad style to use events in tests and you were caught with it. Please, upload file and wait, then it's uploaded, then do the rest.  You also didn't mock the ReductStore client and it won't work properly, it's very important to check that we send data to the database but you don't do it.